### PR TITLE
Kelsha Darby -  User Story 24

### LINKED ISSUE #27: User Story 24, Decreasing Item Quantity from Cart
As a visitor
When I have items in my cart
And I visit my cart
Next to each item in my cart
I see a button or link to decrement the count of items I want to purchase
If I decrement the count to 0 the item is immediately removed from my cart

### PR DESCRIPTION
**Pull request by kelshadarby**
### Items in cart can be decremented and removed if decremented to zero
Closes #27 

**_Test_**
- Added to cart show page test to test that items can be decremented and removed if decremented to zero

**_View_**
- Added link to subtract an item from the cart

**_Model_**
- Add `remove_item` method to cart model
- Add conditional to `subtract_item` method to remove item if the quantity is less than 1

### CODE REVIEW

File: app/controllers/cart_controller.rb [hunk 1]
```diff
@@ -17,10 +17,10 @@ def update
 
     if params[:quantity] == "add" && (cart.contents[item.id.to_s] + 1) <= item.inventory
       cart.add_item(params[:item_id].to_s)
-    elsif params[:quanity] == "subtract"
+    elsif params[:quantity] == "subtract"
```
Comment (brian-greeson) [inline app/controllers/cart_controller.rb:20]: I like the quantity of your spelling mistakes

**Review by brian-greeson — APPROVED** [review]
Changes looks good. Let's get this into master

File: app/views/cart/show.html.erb [hunk 2]
```diff
@@ -31,8 +31,9 @@
       <td>
         <center>
           <p id="item-quantity-<%= item.id %>">
+            <%= link_to "-", update_cart_path(item.id, quantity: :subtract), method: :patch %>
             <%=quantity%>
-            <%= link_to "+", update_cart_path(item_id: item.id, quantity: :add), method: :patch %>
+            <%= link_to "+", update_cart_path(item.id, quantity: :add), method: :patch %>
```
Comment (fredrondina) [inline app/views/cart/show.html.erb:36]: nice change, the link looks much cleaner now

**Review by fredrondina — APPROVED** [review]
Looks Good

**Pull request merged**